compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala

```diff
@@ -642,7 +643,7 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
     val selfText = {
       val selfName = if (self.name == nme.WILDCARD) keywordStr("this") else self.name.toString
       (selfName ~ optText(self.tpt)(": " ~ _) ~ " =>").close
-    } provided !self.isEmpty
+    } provided (!self.isEmpty && !constr.symbol.owner.is(Module))
```

Comment:
Not sure this should go into `RefinedPrinter`. You could reproduce this behavior in `DecompilerPrinter` by replacing the self type with an `EmptyTree`

Reply:
Will move it to `DecompilerPrinter`